processor/src/main/java/org/mapstruct/ap/internal/processor/MethodRetrievalProcessor.java

```diff
@@ -319,6 +342,10 @@ private SourceMethod getMethodRequiringImplementation(ExecutableType methodType,
             .build();
     }
 
+    private List<SubClassMappingOptions> sortSubClassMappings(Set<SubClassMappingOptions> subClassMappingOptions) {
+        return subClassMappingOptions.stream().sorted( new SubClassInheritenceSorter() ).collect( Collectors.toList() );
```

Comment:
True, the current implementation doesn't do any sorting atm, but just converts it into a list. 
I noticed that a `LinkedHashSet` is used, I guess that the reason for this is to maintain the user defined order while keeping it distinct?
With a `Set` my assumption would be that the order does not matter. Wouldn't it then be better to explicitly use the `LinkedHashSet` to identify the collection instead of a `Set`?

Reply:
Made it explicit that the Set is a LinkedHashSet and not a Set with undefined ordering.
Also removed the sorter because it is not needed anymore.